[Xinpeng] IP

subiloble opened the pull request:
## Eureka
> "Eureka! I have found it!" – Archimedes ([source](https://en.wikipedia.org/wiki))

Eureka frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ **SUPER** **FAST** to use 🚀

All you need to do is,

1. Download it from [here](https://github.com/subiloble/ip).
2. **Run** it.
3. **Add** your tasks and let the app handle them. 😄

And it is **FREE!**

### Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 0

Commit:
Level 1

Commit:
Level 2

Commit:
Level 3

Commit:
Level 4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level 6

Commit:
A-Enums

Commit:
Level 7 in branch

Commit:
Level 7 in branch new changes

Commit:
Merge branch 'branch-Level-7'

Commit:
Level 8

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Merge branch 'add-gradle-support'

Commit:
A-Gradle

Commit:
A-JUnit

Commit:
A-Jar

Commit:
A-JavaDoc

Commit:
Level-9

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
A-CodingStandard

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/task/Task.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/task/Task.java
#	src/main/java/ui/Ui.java

Commit:
Level-9

Commit:
A-CheckStyle

Commit:
Merge branch 'A-CheckStyle'

Commit:
A-CheckStyle

Commit:
Merge branch 'branch-A-CheckStyle'

kevinlimantoro123 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,43 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
really clean code!

kevinlimantoro123 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
```
This follows all the coding standard guidelines!

kevinlimantoro123 commented on src/test/java/task/EventTest.java:
```diff
@@ -0,0 +1,57 @@
+
+import org.junit.jupiter.api.Test;
+import java.time.LocalDate;
+
```
comprehensive and clear tests!

kevinlimantoro123 commented on src/main/java/task/ToDo.java:
```diff
@@ -0,0 +1,26 @@
+import java.time.LocalDate;
+
+/** Todo task */
+
```
Everything looks good!

kevinlimantoro123 reviewed (commented):
Overall, this code is really clear and straightforward. I can easily understand what each function does and how everything functions

Commit:
Level-10

Commit:
Merge branch 'branch-Level-10'

Commit:
Update chatbot icon

gn07 commented on src/main/java/eureka/Eureka.java:
```diff
@@ -0,0 +1,335 @@
+package eureka;
+
+import java.time.LocalDate;
```
These imports are well organized!

gn07 commented on src/main/java/eureka/Main.java:
```diff
@@ -0,0 +1,32 @@
+public class Main extends Application {
+
+    private final Eureka eureka = new Eureka();
+
```
i like how clean the code is!

gn07 commented on src/main/java/eureka/Eureka.java:
```diff
@@ -0,0 +1,335 @@
+                break;
+
+            case FIND:
+                if (input.length() < 5) {
```
the conditionals are all put on a separate line, which follows the java coding standard!

gn07 commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,195 @@
+    /** Prints underlines for separation. */
+    public void printLine() {
+        System.out.print("  ");
+        for (int i = 0; i < 50; i++) {
```
the loop is wrapped in curly braces even if theres only 1 line, which complies with java coding standards!

gn07 reviewed (commented):
The code is well formatted and easy to read. It has a lot of comments which helps with understanding the code. I like the design of the logo as well.

Commit:
Add welcome message to dialog box

A welcome message is now displayed in the dialog box when the application starts.
This improves the user experience by providing a friendly greeting and setting
expectations for first-time users.

Commit:
A-Assertion

Commit:
A-CodeQuality

Commit:
Refactor storage class methods

By adding a new private method createTask, SLAP is reinforced in the storage class.

Commit:
Refactor task classes with Stream API

There are several classes inheriting the class Task.

Improving the ToDo, Deadline, Event classes by applying `Optional` to avoid null checks and utilising Stream API where applicable.

subiloble committed:
Merge pull request #1 from subiloble/branch-A-Assertions

A-Assertion

subiloble committed:
Merge pull request #2 from subiloble/branch-A-CodeQuality

Branch a code quality

subiloble committed:
Merge pull request #3 from subiloble/branch-A-Streams

Refactor task classes with Stream API

Commit:
C-Undo

Commit:
A-CI

Commit:
Revert "A-CI"

This reverts commit ddf3a7a724911b396aca83f77053d0a9f798cac6.

Commit:
Fix syntax

subiloble committed:
Merge pull request #4 from subiloble/fix

Fix

Commit:
Fix bugs in undo

Commit:
A-BetterGui

Commit:
A-Personality

Commit:
A-MoreErrorHandling

Commit:
A-MoreTesting

Commit:
A-AiAssisted

Commit:
A-UserGuide

Commit:
Fix the title of the user guide

Commit:
Change README.md file location

Commit:
Create Jar file

Commit:
Reformat classes by checkstyle

Commit:
Fix typos in javafx

Commit:
Update the user guide

Commit:
Update undo handler

Commit:
Update command handler
